PyRDF/Proxy.py

```diff
@@ -62,16 +88,77 @@ def GetValue(self):
         Value of the current action node
             This is the value obtained after executing the
             current action node in the computational graph.
-
         """
-        if not self.action_node.value:  # If event-loop not triggered
-            from . import current_backend
-            generator = CallableGenerator(self.action_node.get_head())
+        from . import current_backend
+        if not self.proxied_node.value:  # If event-loop not triggered
+            generator = CallableGenerator(self.proxied_node.get_head())
             current_backend.execute(generator)
 
-        return self.action_node.value
+        return self.proxied_node.value
 
     def _call_handler(self, *args, **kwargs):
         # Handles an operation call to the current action node
         # and returns result of the current action node.
         return getattr(self.GetValue(), self._cur_attr)(*args, **kwargs)
+
+
+class TransformationProxy(Proxy):
+    """
+    A proxy object to an instantiated node. Used as a controller of the user
+    references to the node itself. When the user deletes reference to a
+    node (e.g. assigning the same variable to another operation), the proxy
+    object will get destroyed by Python, thus flagging the node to be without
+    user references anymore.
+    """
+
+    def __getattr__(self, attr):
+        """
+        Intercepts calls to attributes and methods of the proxied node and
+        returns the appropriate object(s).
+
+        Parameters
+        ----------
+        attr : str
+            The name of the attribute or method of the proxied node the user
+            wants to access.
+        """
+        # Old: some tests call dunder methods on nodes so
+        # proxies have to return dunder methods too.
+        # Check if the current call is a dunder method call
+        # import re
+        # if re.search("^__[a-z]+__$", attr):
+        #     # Raise an AttributeError for all dunder method calls
+        #     raise AttributeError("Such an attribute is not set ! ")
+
+        from . import current_backend
+        # if attr is a supported operation, start
+        # operation and node creation
+        if attr in current_backend.supported_operations:
+            self.proxied_node._cur_attr = attr  # Stores new operation name
+            return self._call_handler
+        else:
+            try:
+                return getattr(self.proxied_node, attr)
+            except AttributeError as e:
+                raise e
```

Comment:
What's the error message that gets printed if we reach this point? Is it worth to raise a custom message?

Reply:
it prints that the attribute is not defined. If the `except` statement is left blank, it would go against PEP8 style rules. In fact, flake8 gives the error `E722 do not use bare 'except'`. If you think we shouldn't raise the error message, we could just print something to let the user know that's not a defined attribute, but I'm not sure that'd be worthwhile

Reply:
Forcing the error gives such a message:
```
>>> a.randomattr
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "PyRDF/Proxy.py", line 128, in __getattr__
    raise e
AttributeError: 'Node' object has no attribute 'randomattr'

```